cmd/tailscale/cli/serve.go

```diff
@@ -7,46 +7,105 @@ package cli
 import (
 	"context"
 	"encoding/json"
+	"errors"
 	"flag"
 	"fmt"
 	"io"
+	"net"
+	"net/url"
 	"os"
+	"path"
+	"path/filepath"
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
 
 	"github.com/peterbourgon/ff/v3/ffcli"
+	"golang.org/x/exp/slices"
 	"tailscale.com/ipn"
+	"tailscale.com/ipn/ipnstate"
+	"tailscale.com/tailcfg"
 	"tailscale.com/util/mak"
+	"tailscale.com/version"
 )
 
 var serveCmd = newServeCommand(&serveEnv{})
 
 // newServeCommand returns a new "serve" subcommand using e as its environmment.
 func newServeCommand(e *serveEnv) *ffcli.Command {
 	return &ffcli.Command{
-		Name:       "serve",
-		ShortHelp:  "TODO",
-		ShortUsage: "serve {show-config|https|tcp|ingress} <args>",
-		LongHelp:   "", // TODO
-		Exec:       e.runServe,
-		FlagSet:    e.newFlags("serve", func(fs *flag.FlagSet) {}),
+		Name:      "serve",
+		ShortHelp: "[ALPHA] Serve from your Tailscale node",
+		ShortUsage: strings.TrimSpace(`
+  serve [flags] <mount-point> {proxy|path|text} <arg>
+  serve [flags] <sub-command> [sub-flags] <args>`),
+		LongHelp: strings.TrimSpace(`
+*** ALPHA; all of this is subject to change ***
+
+The 'tailscale serve' set of commands allows you to serve
+content and local servers from your Tailscale node to
+your tailnet. 
+
+You can also choose to enable the Tailscale Funnel with:
+'tailscale serve funnel on'.
+
+This allows your services to be accessed over the internet,
```

Comment:
It's not clear that this, down in its own paragraph, refers to Funnel and not Serve in general.

Let's move it up to line 51 and say "Funnel" instead of "This".

Reply:
Done